app/src/main/java/com/example/criminalintent/data/local/CrimeDao.kt

```diff
@@ -15,11 +16,20 @@ interface CrimeDao {
     fun getCrimes(): LiveData<List<Crime>>
 
     @Query("SELECT * FROM crime WHERE id =(:id)")
-    fun getCrime(id: UUID): LiveData<Crime?>
+    fun getCrime(id: Int): LiveData<Crime?>
 
     @Update
     fun updateCrime(crime: Crime)
 
     @Insert
     fun addCrime(crime: Crime)
+
+    fun upsertCrime(crime: Crime) {
```

Comment:
Needs to be marked with @Transaction in order to make sure that no changes are done in the db between addCrime and updateCrime
https://stackoverflow.com/a/48641762/2625236